boards/arm/nrf9160_pca10090/nrf9160_pca10090ns_defconfig

```diff
@@ -12,6 +12,8 @@ CONFIG_ARM_TRUSTZONE_M=y
 
 # This Board implies building Non-Secure firmware
 CONFIG_TRUSTED_EXECUTION_NONSECURE=y
+CONFIG_FLASH_LOAD_OFFSET=0x40000
```

Comment:
If this value is based on slot0_partition, than why not get it from DT like we do MCUBOOT?

Reply:
Agree with Kumar, we should do this. It should be based on slot0_nonsecure, if I am not wrong.

Reply:
Can't we take DT_CODE_PARTITION_OFFSET / SIZE ?

Reply:
I had tried: `CONFIG_FLASH_LOAD_OFFSET = DT_CODE_PARTITION_OFFSET`, didn't work.

Reply:
You can do what you did initially, but just put it in the board Kconfig.defconfig for now.  (Take a look at boards/arm/mimxrt1050_evk/Kconfig.defconfig for example).

Reply:
Would it work if you use: $(dt_hex_val,DT_CODE_PARTITION_OFFSET)?

Reply:
Yes that should work

Reply:
Otherwise, sure, we can always add `select FLASH_LOAD_OFFSET=$(dt_hex_val,DT_CODE_PARTITION_OFFSET)` inside Kconfig.defconfig (assuming that _this_ would work) :)

Reply:
> Otherwise, sure, we can always add `select FLASH_LOAD_OFFSET=$(dt_hex_val,DT_CODE_PARTITION_OFFSET)` inside Kconfig.defconfig (assuming that _this_ would work) :)

I doubt `select` work, you can only `select` a boolean entry, afaik. 
Kumar suggestion does work, but the board Kconfig.defconfig is shared between secure and non-secure variants, so we'd need something like:

`config FLASH_LOAD_OFFSET`
`	default $(dt_hex_val,DT_CODE_PARTITION_OFFSET) if CONFIG_TRUSTED_EXECUTION_NONSECURE`

Reply:
The "if" part is acceptable, I think.

 So, `
CONFIG_FLASH_LOAD_OFFSET=FLASH_LOAD_OFFSET=$(dt_hex_val,DT_CODE_PARTITION_OFFSET)` in the defconfig of the non-secure does not work, @lemrey?

Reply:
> So, ` CONFIG_FLASH_LOAD_OFFSET=FLASH_LOAD_OFFSET=$(dt_hex_val,DT_CODE_PARTITION_OFFSET)` in the defconfig of the non-secure does not work, @lemrey?

No:
`CONFIG_FLASH_LOAD_OFFSET=$(dt_hex_val,DT_CODE_PARTITION_OFFSET)`
cannot be placed in `boards/arm/nrf9160_pca10090/nrf9160_pca10090ns_defconfig`

It needs to go in `boards/arm/nrf9160_pca10090/Kconfig.defconfig`

Reply:
@galak Updated as discussed.